sql/expression/auto_increment.go

```diff
@@ -93,30 +86,16 @@ func (i *AutoIncrement) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 	}
 
 	if given != nil && cmp != 0 {
-		// check if the given value is greater than autoIncVal
-		cmp, err := i.Type().Compare(given, i.autoIncVal.value)
-		if err != nil {
-			return nil, err
-		}
-		if cmp <= 0 {
-			// if it's less, return it and don't increment
-			return given, nil
-		}
-		i.autoIncVal = NewLiteral(given, i.Type())
-	}
-
-	val, err := i.autoIncVal.Eval(ctx, row)
-	if err != nil {
-		return nil, err
+		return given, nil
```

Comment:
In general, Eval should be safe to call multiple times, and this change makes that not the case

I think it's fine for now, and you have tests, but add a note for whoever breaks this